### DISCUSSION TITLE
Middleware does not work in Layout component.

### DISCUSSION THREAD

**Discussion by thingsneverchange** [nuxt/nuxt#17584] [Questions]
```
<script setup>
// /layout/mylayout.vue
definePageMeta({
  middleware: "auth"
});
</script>
```
This option seems to work on Page components only. Is there a reason why this doesn't work? I managed to make this work by setting the global prefix to the middleware file, but just curious if this feature is just not yet available, or there's any reason behind it.

**Comment by peterkimzz** [comment]
Got the same issue. Is there any update?

**Comment by tomasSlouka** [comment]
This would be awesome if it would be possible to define a middleware in a layout and not just in pages.
